src/Message.chpl

```diff
@@ -245,6 +262,22 @@ module Message {
             this.size = param_list.size;
         }
 
+        proc getJSON(keys: list(string) = list(string)): string throws {
+            var s: int = if keys.isEmpty() then this.size else keys.size;
+            var json: [0..#s] string;
+            var idx: int = 0;
+            for p in this.param_list {
+                if (keys.isEmpty() || keys.contains(p.key)) {
```

Comment:
saving `keys.isEmpty` into a boolean const would save the recompute

Reply:
Updated.